### PR TITLE
Remove UCR experiment code

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

I was going to upgrade `laboratory`, but noticed it was used in once specific place that seemed pretty niche. The code that used this dependency was all related to the `compare_ucr_dbs` celery task. This task appears to not have been run in the last year at least. I asked Simon if he had any context, and he mentioned that this code was used with ICDS and can likely be removed.

Also removes `json-delta`

### Safety story

Code that is no longer used.

### QA Plan

No QA

### Migrations

- [ ] The migrations in this code can be safely applied first independently of the code
(Removed this section initially because this PR didn't have migrations at first, which was an unfortunate oversight. Reverting to separate out code changes from the migration)

### Rollback instructions

This PR deletes models, so reverting the code without restoring those models would leave us in a state where code is relying on SQL tables that do not exist.

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by calellowitz — APPROVED** [review]

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 1]
```diff
@@ -433,10 +432,6 @@ def generate_rows_and_filters(report_data_cache, report_config, restore_user, ro
     )
     filters_elem = _get_filters_elem(defer_filters, filter_options_by_field, restore_user._couch_user)
 
-    if (report_config.report_id in settings.UCR_COMPARISONS and
-        COMPARE_UCR_REPORTS.enabled(uuid.uuid4().hex, NAMESPACE_OTHER)):
```
Comment (snopoke) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py:437]: can you remove this FF as well

File: corehq/apps/userreports/laboratory/experiment.py [hunk 2]
```diff
@@ -1,67 +0,0 @@
-import json
-import traceback
-
-import json_delta
-import laboratory
-
-from corehq.apps.userreports.models import (
-    ReportComparisonDiff,
-    ReportComparisonException,
-    ReportComparisonTiming,
```
Comment (snopoke) [inline corehq/apps/userreports/laboratory/experiment.py:10]: these models can also be removed

File: corehq/apps/userreports/laboratory/experiment.py [hunk 3]
```diff
@@ -1,67 +0,0 @@
-import json
-import traceback
-
-import json_delta
```
Comment (snopoke) [inline corehq/apps/userreports/laboratory/experiment.py:4]: can this lib also be removed?

File: corehq/apps/userreports/tasks.py [hunk 4]
```diff
@@ -208,77 +200,6 @@ def _iteratively_build_table(config, resume_helper=None, in_place=False, limit=-
             current_config.save()
 
 
-@task(serializer='pickle', queue=UCR_CELERY_QUEUE)
-def compare_ucr_dbs(domain, report_config_id, filter_values, sort_column=None, sort_order=None, params=None):
-    if report_config_id not in settings.UCR_COMPARISONS:
```
Comment (snopoke) [inline corehq/apps/userreports/tasks.py:213]: `settings.UCR_COMPARISONS` can be removed

File: corehq/apps/app_manager/fixtures/mobile_ucr.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/app_manager/fixtures/mobile_ucr.py:437]: 260d92f1e72cc3d2f453d53d7fdd79c3523fca67

File: corehq/apps/userreports/laboratory/experiment.py (hunk 2 above)
Comment (gherceg) [reply corehq/apps/userreports/laboratory/experiment.py:10]: 7a74c523275cfb0d5f49b021e08e3c0f77d31537

File: corehq/apps/userreports/tasks.py (hunk 4 above)
Comment (gherceg) [reply corehq/apps/userreports/tasks.py:213]: 0d1f65a167ec49d37e686487bf6e34d9f4e6a68f

File: corehq/apps/userreports/laboratory/experiment.py (hunk 3 above)
Comment (gherceg) [reply corehq/apps/userreports/laboratory/experiment.py:4]: 13c57e68d03bba8e51ea3282e481002c4e6db9a6

**Comment by gherceg** [comment]
The test failure is real, but I'm having trouble understanding the cause. When running the tests in `corehq.apps.userreports.tests.test_columns.TestExpandedColumn`, on this branch the client returned by `get_redis_client()` is of type `TestRedisClient` which doesn't have the attribute `get_client`, but on master the client returned is of type `RedisCache`.

There is [this code](https://github.com/dimagi/commcare-hq/blob/cb47dc8615015a01534d9948f52041f83708886f/corehq/tests/noseplugins/redislocks.py#L32) which appears to setup the mock for `get_redis_client`, but on master this patch appears to be disengaged for the tests in question. I'm not sure how the changes made here would have any impact on this behavior. Any ideas?

**Comment by snopoke** [comment]
@gherceg I'm not sure what the issue could be. From what I can tell the `RedisLockTimeoutPlugin` is mean to be active for all tests.

Perhaps @millerdev knows?

**Comment by millerdev** [comment]
The test issue appears to have been caused by the removal of this line:

https://github.com/dimagi/commcare-hq/blob/5c453c07a17a2c379b75d018c1003225d296e2ad/corehq/apps/app_manager/fixtures/mobile_ucr.py#L37

which chained to other modules importing `get_redis_client` before `RedisLockTimeoutPlugin` installed its patch. I added a commit that will hopefully fix the issue, although the `real_redis_client` decorator may need to be applied in more places.

**Review by millerdev — APPROVED** [review]

**Comment by gherceg** [comment]
I realized after merging this that this PR does not follow our migration best practices (I had unfortunately removed that section when first creating the PR because it didn't contain migrations to start, and therefore it didn't cross my mind). However, I've given an explanation in the description as to why this is still safe.

> This PR does delete models that code potentially uses which typically would warrant rolling out the code changes separately, however this is safe for multiple reasons. In the scenario where the SQL tables have been deleted, but the code has not yet been applied, the following is true:
> 
> - If the relevant code does run, it is always within a celery task compare_ucr_dbs, which means the task would fail, but it would not cause a user facing error.
> - the compare_ucr_dbs task has not been run in years on any environment, which means the chances of even facing an error in celery is highly unlikely.

**Comment by gherceg** [comment]
Nevermind. There are places outside of the celery task, like domain deletion that reference these tables. It is best to revert and separate out the migration from the code changes.

**Pull request merged**